nemo/collections/llm/recipes/CONFIGURATION-HIERARCHY.md

```diff
@@ -16,135 +16,151 @@
+  check_for_nan_in_grad: bool = False # If true, check for NaNs in gradients _before_ communication collective
+  bucket_size: Optional[int] = None # Maximum number of parameters in each bucket
+  average_in_collective: bool = False # If true, compute average in collective directly, as opposed to dividing by the dp_size first and then computing sum in the collective
+  fp8_param_gather: bool = False # If true, keep the compute param in fp8 (do not use any other intermediate dtype) and perform the param all-gather in fp8
   ```  
   </blockquote>
   </details>
   </blockquote>
   </details>
 
-  <details>
+  <details open>
   <summary>callbacks (Optional[List[Callback]]=None)</summary>
   <blockquote>
 
-  <details><summary><a href="https://github.com/NVIDIA/NeMo/blob/main/nemo/utils/exp_manager.py#L240">TimingCallback</a> (Callback)</summary>
+  <details open><summary><a href="https://github.com/NVIDIA/NeMo/blob/main/nemo/utils/exp_manager.py#L240">TimingCallback</a> (Callback)</summary>
   <blockquote>
 
   ```sh
-  reduction: str = "mean"
-  sync_cuda: bool = False
-  buffer_size: int = 1
+  reduction: str = "mean" # reduction over multiple timings of the same timer
+  sync_cuda: bool = False # if True torch.cuda.synchronize() is called for start/stop
+  buffer_size: int = 1 # if positive, limits the number of stored measures per name
   ```
   </blockquote>
   </details>
 
-  <details><summary><a href="https://github.com/NVIDIA/NeMo/blob/main/nemo/lightning/pytorch/callbacks/megatron_comm_overlap.py">MegatronCommOverlapCallback</a> (Callback)</summary>
+  <details open><summary><a href="https://github.com/NVIDIA/NeMo/blob/main/nemo/lightning/pytorch/callbacks/megatron_comm_overlap.py">MegatronCommOverlapCallback</a> (Callback)</summary>
   <blockquote>
 
   ```sh
-  tp_comm_overlap: bool = None
-  tp_comm_overlap_cfg: TransformerLayerTPOverlapCfg = None
-  overlap_p2p_comm: bool = None
-  batch_p2p_comm: bool = None
-  overlap_grad_reduce: bool = None
-  overlap_param_gather: bool = None
-  overlap_param_gather_with_optimizer_step: bool = None
-  align_param_gather: bool = None
-  bucket_size: int = None
-  defer_embedding_wgrad_compute: bool = None
-  wgrad_deferral_limit: int = None
+  tp_comm_overlap: bool = None # Enable tensor parallel overlap (experimental)
+  tp_comm_overlap_cfg: TransformerLayerTPOverlapCfg = None # Tensor parallel overlap config
+  overlap_p2p_comm: bool = None # Enable pipeline parallel overlap
+  batch_p2p_comm: bool = None # Batch pipeline parallel send/recv into a single op
+  overlap_grad_reduce: bool = None # Overlap data parallel gradient reduction with compute
+  overlap_param_gather: bool = None # Overlap data parallel parameter gather with compute
+  overlap_param_gather_with_optimizer_step: bool = None # Overlap data parallel parameter gather optimizer step
+  align_param_gather: bool = None # Align data parallel parameter gather across virtual pipeline chunks
```

Comment:
All configs defined in `MegatronCommOverlapCallback` override same configs in megatron-lm `ddp`. 

It might be important to note that `MegatronCommOverlapCallback` will override user set config in the same callback when certain conditions are true, e.g. a user sets `align_param_gether=True` for this callback. But, if DP=1, this callback will reset `align_param_gather=False`